File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    n = [0] * 1000
    for e in array:
        n[e] += 1
    t = []
    m = -1
    for i in range(1000):
        if n[i] > m:
            t = [i]
            m = n[i]
        elif n[i] == m:
            t.append(i)
    if len(t) == 1:
        return t[0]
    return -1

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    let n = new Array(1000).fill(0);
    for (let e of array) {
        n[e] += 1;
    }
    
    let t = [];
    let m = -1;
    
    for (let i = 0; i < 1000; i++) {
        if (n[i] > m) {
            t = [i];
            m = n[i];
        } else if (n[i] === m) {
            t.push(i);
        }
    }
    
    if (t.length === 1) {
        return t[0];
    }
    return -1;
}